tests/foreman/cli/test_contentviews.py

```diff
@@ -315,6 +318,180 @@ def test_delete_cv_version(self):
         result = ContentView.version_delete({u'id': version1_id})
         self.assertEqual(result.return_code, 0)
 
+    def test_remove_cv_environment(self):
+        """@Test: Remove content view from lifecycle environment assignment
+
+        @Feature: Content Views
+
+        @Assert: Content view removed from environment successfully
+
+        """
+        new_org = make_org({u'name': gen_alphanumeric()})
+        new_cv = make_content_view({u'organization-id': new_org['id']})
+        ContentView.publish({u'id': new_cv['id']})
+        result = ContentView.info({u'id': new_cv['id']})
+        env = result.stdout['lifecycle-environments'][0]
+
+        result = ContentView.remove({
+            u'id': new_cv['id'],
+            u'environment-ids': env['id'],
+            u'organization-id': new_org['id'],
+        })
+        self.assertEqual(result.return_code, 0)
+        result = ContentView.info({u'id': new_cv['id']})
+        self.assertFalse(result.stdout['lifecycle-environments'])
+
+    def test_remove_cv_env_and_reassign_key(self):
+        """Test: Remove content view environment and re-assign activation key
+        to another environment and content view
+
+        @Feature: Content Views
+
+        @Assert: Activation key re-assigned successfully
+
+        """
+        new_org = make_org({u'name': gen_alphanumeric()})
+        env = [
+            make_lifecycle_environment({u'organization-id': new_org['id']})
+            for _ in range(2)
+        ]
+
+        source_cv = make_content_view({u'organization-id': new_org['id']})
+        ContentView.publish({u'id': source_cv['id']})
+        result = ContentView.info({u'id': source_cv['id']})
+        cvv = result.stdout['versions'][0]
+        result = ContentView.version_promote({
+            u'id': cvv['id'],
+            u'to-lifecycle-environment-id': env[0]['id'],
+        })
+        self.assertEqual(result.return_code, 0)
+
+        destination_cv = make_content_view({u'organization-id': new_org['id']})
+        ContentView.publish({u'id': destination_cv['id']})
+        result = ContentView.info({u'id': destination_cv['id']})
+        cvv = result.stdout['versions'][0]
+        result = ContentView.version_promote({
+            u'id': cvv['id'],
+            u'to-lifecycle-environment-id': env[1]['id'],
+        })
+        self.assertEqual(result.return_code, 0)
+
+        ac_key = make_activation_key({
+            u'name': gen_alphanumeric(),
+            u'organization-id': new_org['id'],
+            u'lifecycle-environment-id': env[0]['id'],
+            u'content-view-id': source_cv['id'],
+
+        })
+        result = ContentView.info({u'id': source_cv['id']})
+        self.assertEqual(result.stdout['activation-keys'][0], ac_key['name'])
+        result = ContentView.info({u'id': destination_cv['id']})
+        self.assertFalse(result.stdout['activation-keys'])
```

Comment:
better to check the length: `self.assertEqual(len(result.stdout['activation-keys']), 0)`. This will give more meaning when someone else is reading the test.